### PR TITLE
Rework GHA package.yaml and linux-release.yaml

### PR DESCRIPTION
**Pull request by adamkpickering**
This pull request makes a couple of changes to the way that the Linux part of package.yaml and linux-release.yaml work:

- The webhook action (distributhor/workflow-webhook@v2) is replaced by a curl call, which enables us to put the URL for the HTTP request directly in the github action. This makes it easier to understand what is going on. It also enables us to use one token, rather than multiple tokens, for all stages, and to eliminate the `OBS_WEBHOOK_URL_*` secrets (which I believe weren't actually "secrets" in the first place).

- We now have .zip files of the format `rancher-desktop-linux-<branch>.zip` uploaded to S3. These can be grabbed by services in packages, where there is one package per branch, so that we can have a development build for each branch. This should come in handy when testing release branches. The branch names must be in the format `main`, or `release-*`. This applies to the dev channel only.

- I introduced some logic in linux-release.yaml to facilitate a scheme to provide multiple versions of RD in the stable channel. This scheme is similar to how the dev channel works (see above), except instead of the OBS package names being of the form `rancher-desktop-linux-<branch>`, the package names are of the form `rancher-desktop-<major_minor>`, where `major_minor` is the major and minor part of the semantic version (for example `0.7` or `1.0`). All that needs to be done to use this in a release is to upload the .zip file from the artifacts in the GHA workflow run that you want to release - linux-release.yaml should take care of everything else.

### CODE REVIEW

**Comment by mattfarina** [comment]
We are going to hold off merging this until Adam returns.

**Review by ericpromislow — APPROVED** [review]
Looks ok, although I could do only a surface check on the changes. All the most recent actions ran successfully, so it looks ok.

File: .github/workflows/linux-release.yaml
Comment (mattfarina) [inline .github/workflows/linux-release.yaml:35]: The `release_zip_name` has the version with the v. But, the actual download URL has the version with the `v` for the tag name but without the `v` for the file name... for example, `https://github.com/rancher-sandbox/rancher-desktop/releases/download/v0.7.1/rancher-desktop-0.7.1-linux.zip`

All the file names have the version without the `v`...
Screen Shot 2022-01-20 at 6 03 23 PM

File: .github/workflows/package.yaml
Comment (mattfarina) [inline .github/workflows/package.yaml:123]: FYI, `zip_name` (formerly `Rancher Desktop-linux.zip`) is a zip containing the zip uploaded to the release. Go download it from a packaged actions run like https://github.com/rancher-sandbox/rancher-desktop/actions/runs/1725878053. The linux build from the action for the tag is something we download, unzip, and then take the zip inside and upload to the release.

**Review by mattfarina — CHANGES_REQUESTED** [review]
Two things occurred to me reviewing this...

1. It looks like the canary builds could go back and forth on which minor version they are for. For example, let's say 1.0.0 is out and is on the release-1.0 branch. A change lands on this branch so a canary is built for it. That would be pushed to OBS. Then a change lands on main, which is development headed for the 1.1.x minor releases. The canary is updated for this. The code on main and the code the release branch will diverge quite a bit. Shouldn't the dev / canary builds just be from `main`?
2. The zip file name that's pulled from the release doesn't match what we have being built.

**Comment by adamkpickering** [comment]
> It looks like the canary builds could go back and forth on which minor version they are for. For example, let's say 1.0.0 is out and is on the release-1.0 branch. A change lands on this branch so a canary is built for it. That would be pushed to OBS. Then a change lands on main, which is development headed for the 1.1.x minor releases. The canary is updated for this. The code on main and the code the release branch will diverge quite a bit. Shouldn't the dev / canary builds just be from main?

The idea is that each OBS package (note that we're talking about OBS packages here, which each represent an available version of Rancher Desktop) is configured to pull down a .zip file with the name of a specific branch in it. So the OBS package `rancher-desktop-main` pulls down `rancher-desktop-linux-main.zip`, the OBS package `rancher-desktop-release-0.7` pulls down `rancher-desktop-linux-release-0.7.zip`, and so on. Note that OBS is not configured this way at the moment - until this PR is merged we need to keep `rancher-desktop-main` pulling `rancher-desktop-canary.zip`. But once this PR is merged I will change it to pull `rancher-desktop-linux-main.zip`, which will work because that zip file will actually be there.

The end result, which I *think* makes sense, is that the dev channel has one bleeding-edge version of RD for each of `main`, `release-0.7`, and so on. This complicates things from the user's point of view because a) they must select the branch they want at install time, rather than just doing `zypper install rancher-desktop`, and b) they cannot update willy-nilly because they may unintentionally update RD to a version from another branch. The benefit, of course, is that people can get the latest build from any branch. It may be too complex. But it *is* the dev branch: people are aware that they are signing up for a little pain. At the very least we should document this. Your call.

File: .github/workflows/linux-release.yaml
Comment (adamkpickering) [reply .github/workflows/linux-release.yaml:1]: Changed to `Upload Linux release`. I'd just go with `Linux release upload` but it is ambiguous (is `release` a noun or a verb?).

File: .github/workflows/package.yaml
Comment (adamkpickering) [reply .github/workflows/package.yaml:123]: Yes, this was something I realized while putting this together. I wonder if we should document this process somewhere. Though it will probably change at some point anyways.

**Comment by adamkpickering** [comment]
Oh also I should communicate another idea: if we are providing a version of RD for the latest commit on each branch, we should probably change the version string in OBS so that it doesn't have any actual version (as in v.X.Y.Z) in it. Having a version will just confuse people because the package they're installing won't have anything to do with that version. Instead, I'd like to put just the branch and the commit.

Of course, it may not be valid to do this for deb/rpm package. But it is something I'd like to look into.

File: .github/workflows/linux-release.yaml
Comment (adamkpickering) [reply .github/workflows/linux-release.yaml:35]: Yes, previous releases have the version without the `v` in the linux .zip file. Here is what I envisioned:

1) `zip_name` is generated using the tag, which is a version string with the`v`
2) the file with the name `zip_name` is uploaded as an artifact
3) when you want to make a release, you simply take the linux .zip file from the workflow run and include it as an asset in the release without changing its name
4) the linux-release.yaml workflow processes the .zip file and its name in order to make things work with OBS

Now that you write this, I realize that having the `v` in the .zip file in a release is inconsistent with the names of the files from other platforms, which isn't great.

I can change this, but it will complicate package.yaml since I will need to process, for example, the `v1.0.0` tag name into `1.0.0`. Alternatively you could just remove the `v` whenever you make a release, but it would be an extra, undocumented, manual step, and I'd still need to change linux-release.yaml.

**Comment by adamkpickering** [comment]
As requested, these are the steps to install Rancher Desktop from the dev channel. First you need to add the repo:

```
curl https://download.opensuse.org/repositories/isv:/Rancher:/dev/deb/Release.key | sudo apt-key add
sudo add-apt-repository 'deb https://download.opensuse.org/repositories/isv:/Rancher:/dev/deb/ ./'
sudo apt update
```

At this point you need to figure out which version to install. You can see available versions of Rancher Desktop with:

```
apt list -a rancher-desktop
```

Once you find the version you want to install you can install it with:

```
sudo apt install rancher-desktop=<version>
```

This works even if you already have a version of rancher desktop installed.

**Comment by adamkpickering** [comment]
Here are the instructions for installing a specific version of RD using zypper from the dev channel. First add the repo:

```
sudo zypper addrepo https://download.opensuse.org/repositories/isv:/Rancher:/dev/rpm/isv:Rancher:dev.repo
sudo zypper refresh
```

Then, list the available versions of RD using:

```
zypper search -s rancher-desktop
```

Finally, install the version you want with:

```
zypper install rancher-desktop-<version>
```

Note that you may have to do `zypper install --oldpackage rancher-desktop-<version>` if you are doing a downgrade.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**
